akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -292,6 +292,7 @@ class TLSActor(
       if (tracing) log.debug("closing inbound")
       try engine.closeInbound()
       catch { case ex: SSLException ⇒ outputBunch.enqueue(UserOut, SessionTruncated) }
+      lastHandshakeStatus = engine.getHandshakeStatus
```

Comment:
So, to further understand what the effect is of this fix. Is it that without that line, the TLSActor gets stuck waiting for connection being closed by the peer? And the peer is waiting for the close_notify which was never sent out?